Realm/RLMResults.mm

```diff
@@ -376,6 +376,9 @@ - (id)objectAtIndexedSubscript:(NSUInteger)index {
 }
 
 - (id)aggregate:(NSString *)property method:(util::Optional<Mixed> (Results::*)(size_t))method methodName:(NSString *)methodName {
+    if (_results.get_mode() == Results::Mode::Empty) {
```

Comment:
Ah, the two methods in question deal with `Mode::Empty` because they need to work with a table prior to calling into `Results` (to prepare the query and sort descriptors respectively), and that table may not exist. So they are consistent with the logic you outlined, but just for a slightly different reason.